src/views/order_list_view.js

```diff
@@ -0,0 +1,113 @@
+import QuoteView from '../views/quote_view';
+import Quote from '../models/quote';
+import TradeHistoryView from '../views/trade_history_view';
+import OrderView from '../views/order_view';
+import Order from '../models/order';
+
+const OrderListView = Backbone.View.extend({
+  initialize(params){
+    this.template = params.template;
+    this.bus = params.bus;
+    this.quotes = params.quotes;
+
+    this.listenTo(this.model, 'update', this.render);
+    this.listenTo(this.bus, 'quote_price_change', this.buySellQuote);
+  },
+  render() {
+    this.$('#orders').empty();
+
+    this.model.each((order) => {
+      const orderView = new OrderView({
+        model: order,
+        template: this.template,
+        tagName: 'li',
+        className: 'order', // orders?
+        bus: this.bus,
+      });
+      this.$('#orders').prepend(orderView.render().$el);
+    });
+    this.newOrderFormRender();
+    return this;
+  },
+  newOrderFormRender() {
+    this.$('#add-order-form select').html('');
+    this.quotes.each((quote) =>{
+      const symbol = quote.get('symbol');
+      this.$('#add-order-form select').append(`<option value="${symbol}">${symbol}</option>`);
+    });
+  },
+  events: {
+    'click button.btn-buy, button.btn-sell': 'addOrder',
+  },
+  addOrder(event) {
+    event.preventDefault();
+    const formData = this.getFormData();
+    console.log(formData);
+    const quotePrice = this.quotes.where({symbol: formData['symbol']})[0].get('price');
+    if (event.target.classList.contains('btn-buy')) {
+      formData['buy'] = true;
+      if (quotePrice <= formData['targetPrice']) {
+        // console.log('targetPrice must be lower than current price');
+        this.updateUser('Sorry! You must enter a number that is lower than the current price ');
+        return
+      }
+    }
+    else {
+      formData['buy'] = false;
+      if (quotePrice >= formData['targetPrice']) {
+        // console.log('targetPrice must be higher than current price ');
+        this.updateUser('Sorry! You must enter a number that is  higher than the current price ');
+        return;
+      }
+    }
+    const newOrder = new Order(formData);
+    console.log(formData);
+    if (!newOrder.isValid()) {
+      newOrder.destroy();
+      this.updateUser(newOrder.validationError);
+      return;
+    }
+    this.model.add(newOrder);
+    this.clearFormData();
+    this.updateUser(`Created new order for:  ${newOrder.get('symbol')} `)
+  },
+  getFormData() {
+    const orderData = {};
+    orderData['symbol'] = this.$('#add-order-form select[name="symbol"]').val();
+    orderData['targetPrice'] = Number(this.$('#add-order-form input[name="price-target"]').val());
+    console.log(orderData);
+    return orderData;
+  },
+  clearFormData() {
+    this.$('#add-order-form input[name="price-target"]').val('');
+  },
+  buySellQuote(quote) {
+    const selectedOrder = this.model.where({symbol: quote.get('symbol')});
+
+    selectedOrder.forEach((order) => {
```

Comment:
Currently, the workflow for executing an open order looks like this:

1. Quote price changes
1. `quote_price_change` event on the bus
1. Handled here by `OrderListView.buySellQuote()`
1. Find `Order`s matching the quote
1. For each one, determine if it needs to be sold

But we can simplify this! The key observation is that when you create the `Order` you must know which `Quote` it's for - you need this information to validate the `Order`. Instead of working through the intermediaries of the bus and the `OrderListView`, each `Order` could listen directly to its `Quote` for price changes. Then the workflow would look like:

1. Quote price changes
1. Event on the quote
1. Handler in `Order`
1. Determine if the order needs to be sold

This eliminates dependencies on the bus and the `OrderListView`. It also simplifies the code, since we don't need to work with a list of `Order`s. If there are multiple `Orders` corresponding to the same quote each will have a separate handler registered, and each will run when the event occurs (steps 3-4).